code/modules/mob/living/carbon/human/human_defense.dm

```diff
@@ -423,4 +423,25 @@ Contains most of the procs that are called when a mob is attacked by something
 	apply_damage(stamina_damage, STAMINA, updating_health = TRUE)
 	if(!ear_deaf)
 		adjust_ear_damage(deaf = stun_duration)  //Deafens them temporarily
-	//Perception distorting effects of the psychic scream
+	//Perception distorting effects of the psychic scream*
+
+/mob/living/carbon/human/attackby(obj/item/I, mob/living/user, params)
+	if(stat != DEAD || I.sharp < IS_SHARP_ITEM_ACCURATE)
+		return ..()
+	if(!internal_organs_by_name["heart"])
+		to_chat(user, span_notice("[src] no longer has a heart"))
+		return
+	if(!HAS_TRAIT(src, TRAIT_UNDEFIBBABLE) && get_ghost())
+		to_chat(user, span_warning("You cannot resolve yourself to destroy [src]'s heart, as [p_they()] can still be saved!"))
+		return
+	to_chat(user, span_notice("You start to remove [src]'s heart, preventing [p_them()] from rising again!"))
```

Comment:
where does this check for zombification + missing admin logging

Reply:
You can do this prevently to undefibbable human, like if you know it was bitten by a zombie before and you cannot resurrect it.

Reply:
this doesnt check for zombium though